packages/core/src/typeschema/validation.ts

```diff
@@ -360,3 +388,67 @@ function checkObjectForNull(obj: Record<string, unknown>, path: string, issues:
     }
   }
 }
+
+function matchesSpecifiedValue(value: TypedValue | TypedValue[], element: ElementValidator): boolean {
+  if (element.pattern && !deepIncludes(value, element.pattern)) {
+    return false;
+  } else if (element.fixed && !deepEquals(value, element.fixed)) {
+    return false;
+  }
+  return true;
+}
+
+function matchDiscriminant(
+  value: TypedValue | TypedValue[] | undefined,
+  discriminator: SliceDiscriminator,
+  slice: SliceDefinition
+): boolean {
+  const element = slice.fields[discriminator.path];
+  switch (discriminator.type) {
+    case 'value':
+    case 'pattern':
+      if (!element) {
+        throw new OperationOutcomeError(
```

Comment:
You can just `throw new Error` rather than `throw new OperationOutcomeError(serverError(...))` - It will end up in the same place.

I'm trying to wrap my head around how we get here.... This would be the case where someone writes a FHIR Profile with an invalid slice expression?  Should that be a 500 or a 400?

Reply:
This really should never happen; this is basically just a safety assertion so we get a better error message for this case if it ever pops up